source/extensions/filters/http/ext_proc/mutation_utils.cc

```diff
@@ -174,40 +182,132 @@ absl::Status MutationUtils::applyHeaderMutations(const HeaderMutation& mutation,
       return absl::InvalidArgumentError("Invalid character in set_headers mutation.");
     }
     const LowerCaseString header_name(sh.header().key());
-    const bool append = PROTOBUF_GET_WRAPPED_OR_DEFAULT(sh, append, false);
-    const auto check_op = (append && !headers.get(header_name).empty()) ? CheckOperation::APPEND
-                                                                        : CheckOperation::SET;
-    auto check_result = checker.check(check_op, header_name, header_value);
-    if (replacing_message && header_name == Http::Headers::get().Method) {
-      // Special handling to allow changing ":method" when the
-      // CONTINUE_AND_REPLACE option is selected, to stay compatible.
-      check_result = CheckResult::OK;
-    }
-    switch (check_result) {
-    case CheckResult::OK:
-      ENVOY_LOG(trace, "Setting header {} append = {}", sh.header().key(), append);
-      if (append) {
-        headers.addCopy(header_name, header_value);
-      } else {
-        headers.setCopy(header_name, header_value);
+
+    if (Runtime::runtimeFeatureEnabled(
```

Comment:
You do not need to use runtime guard to maintain backward compatibility.

I think we can just do below, which can honor "append_action" at same time maintain backward compatibility if "append_action" is empty:

1) If "append_action" has value, then honor "append_action" and ignore "append" value.
2) If "append_action" is empty
     2.1) if "append" has value, then honor "append"
     2.2) if "append" is empty, then honor "append" default value, which is false.

Reply:
Checking if 'append_action' has a  value was my initial approach, unfortunately it didn't pan out. The tests failed, and it always took `APPEND_IF_EXISTS_OR_ADD` path no matter the `action` is true or flase.

When adding a header to HeaderValueOption, the append_action has a default value 0 (which is  APPEND_IF_EXISTS_OR_ADD).

In the mean time, append is boolean, either true or false, and I can not tell whether the application desires `action` option or not.

So we don't know the application actually means to use `append` option or `append_action` option.

Besides using runtime guarding, I think we might have other option. If we redefine `enum HeaderAppendAction` to use code value >=1, it might also work.